### DISCUSSION TITLE
Set specific GPU

### DISCUSSION THREAD

**Discussion by Onyoursix** [openai/whisper#353] [Q&A] [answered]
Is there a way to specify the GPU I want to use? I have a machine with multiple GPUs and I would like to load the model into a specific GPU instead of it defaulting to the first one.

**Comment by jongwook** [comment, marked as answer]
Setting the environment `CUDA_VISIBLE_DEVICES` will work, like:

```
CUDA_VISIBLE_DEVICES=3 whisper audio.wav
```

`--device cuda:3` may also work, but I haven't tested.

**Reply by Aaryan369** [reply]
Tried with `--device cuda:3`, it is successfully able to load the model into the chosen GPU.

**Reply by sk7n4k3d** [reply]
And in python ?

**Reply by glangford** [reply]
@nathan-skynet Try
```python
model = whisper.load_model('large-v2', device='cuda:3')
```
See
https://github.com/openai/whisper/blob/ba3f3cd54b0e5b8ce1ab3de13e32122d0d5f98ab/whisper/__init__.py#L99

**Reply by sk7n4k3d** [reply]
Thanks

**Reply by psp631** [reply]
How to make specific modifications on line 99, thank you for your  help.
